frame/lending/src/lib.rs

```diff
@@ -521,54 +486,44 @@ pub mod pallet {
 		}
 	}
 
+	#[allow(type_alias_bounds)] // false positive
+	pub type CreateInputOf<T: Config> = CreateInput<T::LiquidationStrategyId, T::MayBeAssetId>;
+
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
 		/// Create a new lending market.
-		/// - `origin` : Sender of this extrinsic. (Also manager for new market to be created.)
-		/// - `collateral_asset_id` : AssetId for collateral.
-		/// - `reserved_factor` : Reserve factor of market to be created.
-		/// - `collateral_factor` : Collateral factor of market to be created.
-		/// - `under_collaterized_warn_percent` : warn borrower when loan's collateral/debt ratio
-		///   given percentage short to be under collaterized
+		/// - `origin` : Sender of this extrinsic. Manager for new market to be created. Can pause
+		///   borrow & deposits of assets.
 		#[pallet::weight(<T as Config>::WeightInfo::create_new_market())]
 		#[transactional]
-		#[allow(clippy::too_many_arguments)]
-		pub fn create_new_market(
+		pub fn create_market(
 			origin: OriginFor<T>,
-			borrow_asset_id: <T as Config>::AssetId,
-			collateral_asset_id: <T as Config>::AssetId,
-			reserved_factor: Perquintill,
-			collateral_factor: NormalizedCollateralFactor,
-			under_collaterized_warn_percent: Percent,
-			interest_rate_model: InterestRateModel,
-			liquidator: Option<T::GroupId>,
+			input: CreateInput<T::LiquidationStrategyId, T::MayBeAssetId>,
 		) -> DispatchResultWithPostInfo {
 			let who = ensure_signed(origin)?;
-			let market_config = MarketConfigInput {
-				reserved: reserved_factor,
-				manager: who.clone(),
-				collateral_factor,
-				under_collaterized_warn_percent,
-				liquidator,
-			};
-			let (market_id, vault_id) = Self::create(
-				borrow_asset_id,
-				collateral_asset_id,
-				market_config,
-				&interest_rate_model,
-			)?;
-			Self::deposit_event(Event::<T>::NewMarketCreated {
+			let (market_id, vault_id) = Self::create(who.clone(), input.clone())?;
+			Self::deposit_event(Event::<T>::MarketCreated {
 				market_id,
 				vault_id,
 				manager: who,
-				borrow_asset_id,
-				collateral_asset_id,
-				reserved_factor,
-				collateral_factor,
+				input,
 			});
 			Ok(().into())
 		}
 
+		#[pallet::weight(<T as Config>::WeightInfo::create_new_market())]
+		#[transactional]
+		pub fn update_market(
+			origin: OriginFor<T>,
+			_input: UpdateInput<T::LiquidationStrategyId>,
+		) -> DispatchResultWithPostInfo {
+			let _who = ensure_signed(origin)?;
+			// 1. validate owner
+			// 2. update configuration
+			// 3. what is owner updates and forces liquidations? ok.
+			Err(DispatchError::Other("not implemented").into())
+		}
+
```

Comment:
I think it would be nice to introduce `TODO: small-description - clickup-ticket` when we just deploy a stub.